### PR TITLE
Add an ASP.NET Core app to the Windows IIS zero-config test

### PR DESCRIPTION
**Pull request by pjanotti**
Due to the size and time to launch the Windows images the test is run together with the ASP.NET app.

### CODE REVIEW

File: tests/zeroconfig/windows/windows_iis_test.go [hunk 1]
```diff
@@ -86,17 +86,15 @@ func TestWindowsIISInstrumentation(t *testing.T) {
...
-	httpClient := &http.Client{}
-	req, err := http.NewRequest("GET", "http://localhost:8000/aspnetfxapp/api/values", nil)
-	require.NoError(t, err)
-	resp, err := httpClient.Do(req)
-	require.NoError(t, err)
-	defer resp.Body.Close()
-	require.Equal(t, http.StatusOK, resp.StatusCode)
+	requireHTTPGetRequestSuccess(t, "http://localhost:8000/aspnetfxapp/api/values/4")
 	t.Log("ASP.NET HTTP Request succeeded")
+	expectedResourceTraces, err := telemetry.LoadResourceTraces("expected_aspnetfxapp_traces.yaml")
```
Comment (rmfitzpatrick) [inline tests/zeroconfig/windows/windows_iis_test.go:91]: There's yet to be `testutils.Testcase` helper adoption but there's a convention to have a `testdata/resource_<type>` directory where [the loader takes care of some validation](https://github.com/signalfx/splunk-otel-collector/blob/main/tests/testutils/testcase.go#L110). Until a better option system is implemented that more readily accepts arbitrary otlp sink endpoints there's not a real need for it in this suite imo.

In the meantime do you mind moving these to something like `testdata/resource_spans/aspnetfx.yaml` and `testdata/resource_spans/aspnetcore.yaml` for consistency?

**Review by rmfitzpatrick — APPROVED** [review]
lgtm w/ nit

File: tests/zeroconfig/windows/windows_iis_test.go (hunk 1 above)
Comment (pjanotti) [reply tests/zeroconfig/windows/windows_iis_test.go:91]: just for consistency with the type I will use `testdata/resource_traces` (that issue that pdata uses "traces").

Comment (pjanotti) [reply tests/zeroconfig/windows/windows_iis_test.go]: Fixed via 7de808941ab7e67e7c0c0319764454c3a9a5ddd0

**Comment by pjanotti** [comment]
Failures seem unrelated to the change:

- https://github.com/signalfx/splunk-otel-collector/actions/runs/4887091653/jobs/8723604411?pr=3063#step:4:106
- https://github.com/signalfx/splunk-otel-collector/actions/runs/4887091653/jobs/8723604411?pr=3063#step:6:15
- https://github.com/signalfx/splunk-otel-collector/actions/runs/4887091653/jobs/8723604411?pr=3063#step:8:157
- https://github.com/signalfx/splunk-otel-collector/actions/runs/4887091653/jobs/8723544232#step:4:2046

**Pull request merged**